src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -1062,7 +1074,9 @@ private VariantContext subsetRecord(final VariantContext vc, final boolean prese
             for (final Genotype genotype : newGC) {
                 //Set genotype to no call if it falls in the fraction.
                 if (fractionGenotypes > 0 && randomGenotypes.nextDouble() < fractionGenotypes) {
-                    genotypes.add(new GenotypeBuilder(genotype).alleles(diploidNoCallAlleles).noGQ().make());
+                    final List<Allele> noCallAlleles = (genotype.getPloidy() == 2 ? diploidNoCallAlleles :
```

Comment:
the same is duplicated in two places in this file - suggests an extract method is in order: `final List<Allele> noCallAlleles = (g.getPloidy() == 2 ? diploidNoCallAlleles : GATKVariantContextUtils.noCallAlleles(g.getPloidy()));`